crates/matrix-sdk-ui/src/room_list_service/mod.rs

```diff
@@ -112,6 +113,13 @@ pub struct RoomListService {
 }
 
 impl RoomListService {
+    /// Size of the room's ring buffer.
+    ///
+    /// This number should be high enough so that navigating to a room
+    /// previously visited is almost instant, but also not too high so as to
+    /// avoid exhausting memory.
+    const ROOM_OBJECT_CACHE_SIZE: usize = 16;
```

Comment:
That seems to quite low, no? Some bigger tablets or desktop clients might fit that many rooms into the view.

How about ~100? Or 128 if you want it to remain a power of 2.